### PR TITLE
Alliesbranch2

### PR DESCRIPTION
**Pull request by alemmond**
Created a file that reads an array list of data points with known and unknown temperatures and returns an array list of data points with known and interpolated temperatures.
Created a test that checks for correct interpolated values and verifies that all data points in the array list have a temperature value.

### CODE REVIEW

**Review by sechols223 — CHANGES_REQUESTED** [review]
Change

**Comment by sechols223** [comment]
PR name is questionable

**Comment by alemmond** [comment]
updated to account for depth as well as to be able to read 3d array list as input

File: src/main/java/com/github/gissuite/gribinterpolation/core/InverseWeightedDataInterpolator.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                .flatMap(ArrayList::stream)
+                .flatMap(ArrayList::stream)
+                .filter(dataPoint -> Float.isNaN(dataPoint.getTemperatureK()))
+                .collect(Collectors.toList());
+
+        ArrayList<DataPoint> knownDataPoints = (ArrayList<DataPoint>) dataPoints.stream()
+                .flatMap(ArrayList::stream)
+                .flatMap(ArrayList::stream)
+                .filter(dataPoint -> !Float.isNaN(dataPoint.getTemperatureK()))
+                .collect(Collectors.toList());
```
Comment (sechols223) [inline src/main/java/com/github/gissuite/gribinterpolation/core/InverseWeightedDataInterpolator.java:30]: Good job :)

File: src/test/java/com/github/gissuite/gribinterpolation/InverseWeightedDataInterpolatorTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            dataPoints.add(innerList1);
+        }
+        InverseWeightedDataInterpolator.DataInterpolator(dataPoints);
+
+
+        for (ArrayList<ArrayList<DataPoint>> dataPoints1 : dataPoints){
+            for (ArrayList<DataPoint> points : dataPoints1){
+                for (DataPoint point : points ){
+                    for (int k = 0; k < innerList1.size(); k++) {
+                        System.out.print("(" + point.getLatitude() + ", " + point.getLongitude() + ", " + point.getTemperatureK() + ", " + point.getDepth() + ") ");
```
Comment (sechols223) [inline src/test/java/com/github/gissuite/gribinterpolation/InverseWeightedDataInterpolatorTest.java:48]: Don't print in unit tests

File: src/test/java/com/github/gissuite/gribinterpolation/InverseWeightedDataInterpolatorTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+        for (DataPoint dataPoint : innerList2) {
+                Assertions.assertNotEquals(dataPoint.getTemperatureK(), NaN);
+        }
+
+
+
+
+        //check that InverseWeightedDataInterpolator returns the correct data points
```
Comment (sechols223) [inline src/test/java/com/github/gissuite/gribinterpolation/InverseWeightedDataInterpolatorTest.java:66]: Remove commented code

**Review by sechols223 — CHANGES_REQUESTED** [review]
Make changes

File: src/main/java/com/github/gissuite/gribinterpolation/core/InverseWeightedDataInterpolator.java (hunk 1 above)
Comment (alemmond) [reply src/main/java/com/github/gissuite/gribinterpolation/core/InverseWeightedDataInterpolator.java:30]: that was hard to figure out the 3d part

File: src/test/java/com/github/gissuite/gribinterpolation/InverseWeightedDataInterpolatorTest.java (hunk 2 above)
Comment (alemmond) [reply src/test/java/com/github/gissuite/gribinterpolation/InverseWeightedDataInterpolatorTest.java:48]: oh right i just did this to double check sorry!

File: src/test/java/com/github/gissuite/gribinterpolation/InverseWeightedDataInterpolatorTest.java (hunk 3 above)
Comment (alemmond) [reply src/test/java/com/github/gissuite/gribinterpolation/InverseWeightedDataInterpolatorTest.java:66]: oops lol

File: src/main/java/com/github/gissuite/gribinterpolation/core/InverseWeightedDataInterpolator.java (hunk 1 above)
Comment (sechols223) [reply src/main/java/com/github/gissuite/gribinterpolation/core/InverseWeightedDataInterpolator.java:30]: I know and I'm happy your were able to figure it out

**Review by sechols223 — APPROVED** [review]

**Pull request merged**